src/storage/errors.rs

```diff
@@ -407,10 +407,42 @@ pub fn extract_key_error(err: &Error) -> kvrpcpb::KeyError {
         }
         // failed in commit
         Error(box ErrorInner::Txn(TxnError(box TxnErrorInner::Mvcc(MvccError(
-            box MvccErrorInner::TxnLockNotFound { .. },
+            box MvccErrorInner::TxnLockNotFound {
+                start_ts,
+                commit_ts,
+                key,
+                mvcc_info,
+            },
         ))))) => {
-            warn!("txn conflicts"; "err" => ?err);
-            key_error.set_retryable(format!("{:?}", err));
+            // use an error without mvcc_info to construct error the error message
+            let err_without_mvcc = &Error::from(TxnError::from(MvccError::from(
+                MvccErrorInner::TxnLockNotFound {
+                    start_ts: *start_ts,
+                    commit_ts: *commit_ts,
+                    key: key.clone(),
+                    mvcc_info: None,
+                },
+            )));
+
+            warn!("txn conflicts"; "err" => ?err_without_mvcc);
+            key_error.set_retryable(format!("{:?}", err_without_mvcc));
+            let mut txn_lock_not_found = kvrpcpb::TxnLockNotFound::default();
+            txn_lock_not_found.set_key(key.clone());
+            key_error.set_txn_lock_not_found(txn_lock_not_found);
+            // if mvcc_info is present, fill the debug_info for client's further debugging.
+            if let Some(mvcc_info) = mvcc_info {
+                // remove the values in default CF to reduce the size of the response.
+                let mut mvcc_info = mvcc_info.clone();
```

Comment:
Could this `clone` be saved and just consume the original `mvcc_info`?

Reply:
One way is:

```
if let Some(mvcc_info) = mvcc_info.take() {
    ...
```

However, this will remove the MVCC info from the original error. If the outside code want to use the input error again , this info will be lost.